### DISCUSSION TITLE
Reference token

### DISCUSSION THREAD

**Discussion by DaiNguyenTM** [soofstad/react-oauth2-pkce#154] [Q&A] [answered]
Hello everyone,
Could you please confirm whether it works with reference tokens?
Thank you very much,
Dai

**Comment by soofstad** [comment]
Hi Dai,
I don't know what you mean by "reference tokens". Could you please elaborate? :slightly_smiling_face:

**Reply by DaiNguyenTM** [reply]
Hi,

There are 2 types of tokens: self-contained tokens and reference tokens (also called opaque tokens). 
Please take a look at: [IdentityServer4 Documentation](https://identityserver4.readthedocs.io/en/latest/topics/reference_tokens.html).
The reference token is simply a GUID string and needs to be introspected, for exemple : af6070da-310f-43ad-91b6-6516374a156d.

Thank you,
D.

**Comment by soofstad** [comment, marked as answer]
Right, well OAuth2, and this library, does not care what the content of the token is. It's just a string, and does not have to be a JWT. As long as the "reference token" is received, and expected to be sent, in the same way as JWTs, it will work.
